File: python/Vehicle.py
class Vehicle:
    def __init__(self):
        self.PlatNomor = ""
        self.Merk = ""
        self.TahunProduk = ""
        self.Warna = ""
        
    def __init__(self, PlatNomor="", Merk="", TahunProduk="", Warna=""):
        """
        Konstruktor untuk kelas Kendaraan.

        Argumen:
            PlatNomor (str): Nomor plat kendaraan. Defaultnya kosong.
            Merk (str): Merek kendaraan. Defaultnya kosong.
            TahunProduk (str): Tahun produksi kendaraan. Defaultnya kosong.
            Warna (str): Warna kendaraan. Defaultnya kosong.
        """
        self.PlatNomor = PlatNomor
        self.Merk = Merk
        self.TahunProduk = TahunProduk
        self.Warna = Warna
    
    def getPlatNomor(self):
        # Metode getter untuk mendapatkan nomor plat kendaraan.
        return self.PlatNomor
    
    def setPlatNomor(self, PlatNomor):
        # Metode setter untuk mengatur nomor plat kendaraan.
        self.PlatNomor = PlatNomor
        
    def getMerk(self):
        # Metode getter untuk mendapatkan merek kendaraan.
        return self.Merk
    
    def setMerk(self, Merk):
        # Metode setter untuk mengatur merek kendaraan.
        self.Merk = Merk
        
    def getTahunProduk(self):
        # Metode getter untuk mendapatkan tahun produksi kendaraan.
        return self.TahunProduk
    
    def setTahunProduksi(self, TahunProduk):
        # Metode setter untuk mengatur tahun produksi kendaraan.
        self.TahunProduk = TahunProduk
    
    def getWarna(self):
        # Metode getter untuk mendapatkan warna kendaraan.
        return self.Warna
    
    def setWarna(self, Warna):
        # Metode setter untuk mengatur warna kendaraan.
        self.Warna = Warna

File: CPP/Vehicle.cpp
#include "header.hh"

Vehicle::Vehicle() 
{
    this->PlatNomor = "";
    this->Merk = "";
    this->TahunProduk = "";
    this->Warna = "";
}

Vehicle::Vehicle(string PlatNomor, string Merk, string TahunProduk, string Warna) 
{
    this->PlatNomor = PlatNomor;
    this->Merk = Merk;
    this->TahunProduk = TahunProduk;
    this->Warna = Warna;
}

string Vehicle::getPlatNomor()  
{ 
    return this->PlatNomor; 
}

void Vehicle::setPlatNomor(string platNomor) 
{ 
    this->PlatNomor = platNomor; 
}

string Vehicle::getMerk()  
{ 
    return this->Merk; 
}

void Vehicle::setMerk(string merk) 
{ 
    this->Merk = merk; 
}

string Vehicle::getTahunProduk()  
{ 
    return this->TahunProduk; 
}

void Vehicle::setTahunProduk(string tahunProduk) 
{ 
    this->TahunProduk = tahunProduk; 
}

string Vehicle::getWarna()  
{ 
    return this->Warna; 
}

void Vehicle::setWarna(string warna) 
{ 
    this->Warna = warna; 
}

Vehicle::~Vehicle() {}
